what4/src/What4/Solver/Z3.hs

```diff
@@ -235,3 +243,83 @@ instance OnlineSolver (SMT2.Writer Z3) where
       timeout feat (Just z3StrictParsing) mbIOh sym
 
   shutdownSolverProcess = SMT2.shutdownSolver Z3
+
+runZ3Horn ::
```

Comment:
I'd appreciate some more Haddocks about the `Z3Horn` family of functions here. In particular:

* How do the `Horn` functions differ from the normal Z3 functions?
* What is the difference between the Horn functions here in the SyGuS CVC5 functions in `What4.Solver.CVC5`? Is one more powerful than the other?